spec/unit/puppet/type/scheduled_task_spec.rb

```diff
@@ -114,7 +114,7 @@
         :title        => 'Foo',
         :command      => 'C:\Windows\System32\notepad.exe',
         :compatibility => 1,
-      )[:compatibility]).to eq(:'1')
```

Comment:
With the munging in place, this had to be fixed. Do you remember where this value came from / why it happens, @Iristyle?

Is it related to [this note from the docs](
https://puppet.com/docs/puppet/5.3/custom_types.html#properties)?

> If you specify valid values, then Puppet will only accept those values, and it will automatically handle accepting either strings or symbols.

Reply:
https://puppet.com/docs/puppet/5.5/custom_types.html#validation-and-munging

> The default validate method looks for values defined using newvalues and if there are any values defined it accepts only those values (this is exactly how allowed values are validated). The default munge method converts any values that are specifically allowed into symbols. If you override either of these methods, note that you lose this value handling and symbol conversion, which you’ll have to call super for.
> 
> Values are always validated before they’re munged.
> 
> Lastly, validation and munging only* happen when a value is assigned. They have no role to play at all during use of a given value, only during assignment.

Reply:
A lot of the `newvalue` tests are in https://github.com/puppetlabs/puppet/blob/master/spec/unit/property_spec.rb#L230-L397

So yes, the default munge will `to_symbol` the values instead of allowing them to remain as integers, which is something I did not previously realize.  It's probably more idiomatic Puppet then to stick with an integer definition and a `validate` block that tests the range, rather than the enumerated values... but I'm going to test a few things locally.